src/ifcopenshell-python/ifcopenshell/validate.py

```diff
@@ -365,7 +412,61 @@ def validate(f, logger, express_rules=False):
 
         if not has_invalid_value:
             for i, (attr, val, is_derived) in enumerate(zip(attrs, values, entity.derived())):
-
+                try:
```

Comment:
Why is a try catch statement used here? Catching all subtypes of Exception silences a lot of other issues. Please rewrite to proper if statement.